src/main/java/uk/gov/digital/ho/hocs/workflow/api/WorkflowService.java

```diff
@@ -114,11 +116,31 @@ public GetCaseResponse getAllCaseStages(UUID caseUUID) {
 
         GetCaseworkCaseDataResponse inputResponse = caseworkClient.getFullCase(caseUUID);
 
-        Set<SchemaDto> schemaDtos = infoClient.getSchemasForCaseType(inputResponse.getType());
+        GetAllStagesForCaseResponse allStagesForCase = caseworkClient.getAllStagesForCase(caseUUID);
```

Comment:
An 'ok' solution, I think it would be better for performance if we only fetched from 'casework' once. Pulling case data + list of visited stages in one call (more comments underneath).  As SchemaDto already has stageType property we could then filter it out here without making any changes in info service for schema retrieval. 
Current setup forces the case data to be read twice especially with no dedicated method in casework. To make it a little bit quicker we could have created a new method in casework that only fetched list of stageTypes the case has been through.

Reply:
Just to make things clear, this is just advice on how we could have done it slightly neater, no need to action this item at the moment:)